src/hooks/useDataWithPct.js

```diff
@@ -0,0 +1,10 @@
+import { sum } from "d3-array";
+
+export default function useDataWithPct(data) {
+  // calculate percentages for display
+  const totalValue = sum(data.map(({ value }) => value));
+  return data.map((record) => ({
+    ...record,
+    pct: record.value / totalValue,
+  }));
+}
```

Comment:
yeah good point ... this isn't really doing anything React-specific, though I guess I thought it might be at some point which is why I made it a hook. Probably I will change it